File: app/src/main/java/com/dimatest/movieapp/network/entity/MovieResponse.java
package com.dimatest.movieapp.network.entity;

import com.dimatest.movieapp.ui.movieList.models.MoviePagedData;
import com.squareup.moshi.Json;

import java.util.List;

public class MovieResponse {

    @Json(name = "results")
    private List<Movie> result;
    @Json(name = "page")
    private Integer page;
    @Json(name = "total_results")
    private Long totalResults;
    @Json(name = "total_pages")
    private Integer totalPages;

    public MovieResponse() {
    }

    public MovieResponse(List<Movie> result, Integer page, Long totalResults, Integer totalPages) {
        this.result = result;
        this.page = page;
        this.totalResults = totalResults;
        this.totalPages = totalPages;
    }

    public List<Movie> getResult() {
        return result;
    }

    public void setResult(List<Movie> result) {
        this.result = result;
    }

    public Integer getPage() {
        return page;
    }

    public void setPage(Integer page) {
        this.page = page;
    }

    public Long getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(Long totalResults) {
        this.totalResults = totalResults;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Integer totalPages) {
        this.totalPages = totalPages;
    }

    public MoviePagedData mapToPagedData() {
        return new MoviePagedData(page, totalResults, totalPages);
    }
}

File: app/src/main/java/com/dimatest/movieapp/network/entity/MovieResponse.kt
package com.dimatest.movieapp.network.entity

import com.dimatest.movieapp.ui.movieList.models.MoviePagedData
import com.dimatest.movieapp.utils.safeLet
import com.squareup.moshi.Json

data class MovieResponse(
        @field:Json(name = "results")
    var result: List<Movie>? = null,

        @field:Json(name = "page")
    var page: Int? = null,

        @field:Json(name = "total_results")
    var totalResults: Long? = null,

        @field:Json(name = "total_pages")
    var totalPages: Int? = null
) {

    fun mapToPagedData() = safeLet(page, totalResults, totalPages) { page, totalResults, totalPages ->
        MoviePagedData(page, totalResults, totalPages)
    } ?: MoviePagedData(0,0,0)
}